components/Modals/AddGameMasterModal.tsx

```diff
@@ -0,0 +1,240 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { isAddress, parseAbi } from 'viem';
+import { Address, usePublicClient, useWalletClient } from 'wagmi';
+
+import { TransactionPending } from '@/components/TransactionPending';
+import { useGame } from '@/contexts/GameContext';
+import { waitUntilBlock } from '@/hooks/useGraphHealth';
+import { useToast } from '@/hooks/useToast';
+
+type AddGameMasterModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+export const AddGameMasterModal: React.FC<AddGameMasterModalProps> = ({
+  isOpen,
+  onClose,
+}) => {
+  const { game, isAdmin, reload: reloadGame } = useGame();
+
+  const { data: walletClient } = useWalletClient();
+  const publicClient = usePublicClient();
+  const { renderError } = useToast();
+
+  const [newGameMaster, setNewGameMaster] = useState<string>('');
+
+  const [showError, setShowError] = useState<boolean>(false);
+  const [isAdding, setIsAdding] = useState<boolean>(false);
+  const [txHash, setTxHash] = useState<string | null>(null);
+  const [txFailed, setTxFailed] = useState<boolean>(false);
+  const [isSyncing, setIsSyncing] = useState<boolean>(false);
+  const [isSynced, setIsSynced] = useState<boolean>(false);
+
+  const invalidGameMasterAddress = useMemo(() => {
+    return !isAddress(newGameMaster.trim());
+  }, [newGameMaster]);
+
+  const alreadyGameMaster = useMemo(() => {
+    return game?.masters.some(
+      master => master === newGameMaster.trim().toLowerCase(),
+    );
+  }, [game, newGameMaster]);
+
+  const hasError = useMemo(() => {
+    return !newGameMaster || invalidGameMasterAddress || alreadyGameMaster;
+  }, [alreadyGameMaster, newGameMaster, invalidGameMasterAddress]);
+
+  useEffect(() => {
+    setShowError(false);
+  }, [newGameMaster]);
+
+  const resetData = useCallback(() => {
+    setNewGameMaster('');
+
+    setShowError(false);
+    setIsAdding(false);
+    setTxHash(null);
+    setTxFailed(false);
+    setIsSyncing(false);
+    setIsSynced(false);
+  }, []);
+
+  useEffect(() => {
+    if (!isOpen) {
+      resetData();
+    }
+  }, [resetData, isOpen]);
+
+  const onAddGameMaster = useCallback(
+    async (e: React.FormEvent<HTMLDivElement>) => {
+      e.preventDefault();
+
+      if (hasError) {
+        setShowError(true);
+        return;
+      }
+
+      try {
+        if (!walletClient) throw new Error('Could not find a wallet client');
+        if (!(game && game.gameMasterHatEligibilityModule))
+          throw new Error('Missing game data');
+        if (!isAdmin) throw new Error('You are not a game admin');
+
+        setIsAdding(true);
+
+        const transactionhash = await walletClient.writeContract({
+          chain: walletClient.chain,
+          account: walletClient.account?.address as Address,
+          address: game.gameMasterHatEligibilityModule as Address,
+          abi: parseAbi([
+            'function addEligibleAddresses(address[] calldata _addresses) external',
```

Comment:
Doesn't this require another transaction for minting the game master hat?

Reply:
Ah, ya, forgot to mention that this assumes adding eligibility and minting a new hat will be wrapped into a single function. Maybe for now, I'll just link to the hats dashboard after the first transaction is complete

Reply:
Added in bc64c93db8a8398f5e6942b507d76b90e7a973f6